src/extensions/operations/remove-banned-users-cron-handler/src/actions/remove-banned-users.ts

```diff
@@ -0,0 +1,25 @@
+
+import Bluebird from 'bluebird';
+import { OperationContext } from '@directus/types';
+import { getDirectusUsers, deleteUser, deleteCredits } from '../repositories/directus.js';
+import { getGithubUser } from '../repositories/github.js';
+
+export const removeBannedUsers = async (context: OperationContext) => {
+	const users = await getDirectusUsers(context);
+	const removedIds = await Bluebird.map(users, async (user) => {
+		if (!user.external_identifier) {
+			return null;
+		}
+
+		const githubUser = await getGithubUser(user.external_identifier, context);
+
+		if (githubUser === null) {
+			const id = await deleteUser(user, context);
+			await deleteCredits(user, context);
+			return id;
+		}
+
+		return null;
+	}, { concurrency: 8 });
```

Comment:
Let's use just `2` here. The API doesn't like getting too many concurrent requests and starts returning 429 quickly in such cases. We'll also need to make some improvements to keep this working as we get more users, but I'll open a separate issue for that.